solidity/dashboard/src/sagas/liquidity-rewards.js

```diff
@@ -82,6 +94,62 @@ export function* watchFetchLiquidityRewardsData() {
   )
 }
 
+export function* watchLiquidityRewardsEarnedNotification() {
+  for (const [pairName, value] of Object.entries(LIQUIDITY_REWARD_PAIRS)) {
+    yield takeLatest(
+      `liquidity_rewards/${pairName}_liquidity_rewards_earned_notification`,
+      processLiquidityRewardEarnedNotification
+    )
+  }
+}
+
+function* processLiquidityRewardEarnedNotification(action) {
+  const contracts = yield getContractsContext()
+  const displayMessage = action.payload.displayMessage
+
+  const liquidityRewardPairName = action.payload.liquidityRewardPairName
+  const liquidityRewardPair = LIQUIDITY_REWARD_PAIRS[liquidityRewardPairName]
+
+  const LPRewardsContract = contracts[liquidityRewardPair.contractName]
+  const { liquidityRewards } = yield select()
+  const lastNotificationRewardAmount = new BigNumber(
+    liquidityRewards[
+      liquidityRewardPairName
+    ].liquidityRewardsEarnedNotification.lastNotificationRewardAmount
+  )
+  const currentReward = yield call(
+    fetchRewardBalance,
+    action.payload.address,
+    LPRewardsContract
+  )
+  // show the notification if the rewardBalance from LPRewardsContract is greater
+  // than the reward amount that was last time the notification was displayed
+  if (gt(currentReward, lastNotificationRewardAmount)) {
+    if (displayMessage) {
+      yield put(
+        showMessage({
+          type: messageType.REWARD,
+          title: `[${liquidityRewardPairName}] You've earned rewards!`,
+          sticky: true,
+          content: <ViewYourLiquidityBalance />,
```

Comment:
I believe the redux shouldn't know about UI. We should handle displaying the specific message on the UI side and we can use `type` field for this.  An example of how we can do this -> https://stackoverflow.com/questions/35623656/how-can-i-display-a-modal-dialog-in-redux-that-performs-asynchronous-actions

Reply:
I'm not sure if you wanted me to do it exactly like in the first post from your StackOverlow link, but i just moved the `content` variable to the `messageType` constant in `Message.jsx` - that way we are displaying specific content component based on the type, like you proposed.